packages/mobile/src/components/rate-cta-drawer/RateCtaDrawer.tsx

```diff
@@ -0,0 +1,139 @@
+import { View } from 'react-native'
+import InAppReview from 'react-native-in-app-review'
+
+import IconHeart from 'app/assets/images/iconHeart.svg'
+import IconThumbsDown from 'app/assets/images/iconThumbsDown.svg'
+import IconThumbsUp from 'app/assets/images/iconThumbsUp.svg'
+import { Button, Text } from 'app/components/core'
+import { NativeDrawer } from 'app/components/drawer'
+import { useAsyncStorage } from 'app/hooks/useAsyncStorage'
+import { makeStyles } from 'app/styles'
+import { useThemeColors } from 'app/utils/theme'
+
+import { ASYNC_STORAGE_KEY } from './constants'
+
+const DRAWER_NAME = 'RateCallToAction'
+
+const messages = {
+  drawerTitle: 'Enjoying the App?',
+  cancelText: 'No',
+  confirmText: 'Yes!',
+  enjoyingText: 'Are You Enjoying Audius Music?',
+  thankYouText: 'Thanks For The Feedback!'
+}
+
+const useStyles = makeStyles(({ palette, typography, spacing }) => ({
+  contentContainer: {
+    paddingHorizontal: spacing(4),
+    paddingBottom: spacing(4)
+  },
+  title: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingTop: spacing(2),
+    paddingBottom: spacing(6)
+  },
+  titleIcon: {
+    position: 'relative',
+    top: 7,
+    color: palette.neutral,
+    marginRight: spacing(3)
+  },
+  titleText: {
+    textTransform: 'uppercase',
+    marginTop: spacing(4)
+  },
+  buttonRoot: {
+    marginTop: spacing(4)
+  },
+  buttonIcon: {
+    height: 18,
+    width: 18,
+    marginRight: spacing(2)
+  },
+  text: {
+    textAlign: 'center',
+    fontSize: typography.fontSize.large
+  }
+}))
+
+export const RateCtaDrawer = () => {
+  const styles = useStyles()
+  const { neutralLight2 } = useThemeColors()
+  const [userRateResponse, setUserRateResponse] = useAsyncStorage(
```

Comment:
im glad that this hook worked for you. let me know if you have feedback for it.

Reply:
Just had to update so that I could set the value back to null. Other than that it worked well for me